### PR TITLE
Fix display issues when listing coins from a CAT wallet

### PR DESCRIPTION
**Pull request by emlowe**
When using `chia wallet coins list` on a CAT wallet, the display would indicate the amount of the coins in XCH/TXCH and then apply the CAT conversion to the mojos, so you would get strange output like

```
Coin ID: 0xa7a087b1d3f49937dcada65f343a0797ebc75c39176c58629689e426355a1baa
	Address: txch1fckd972smrl3pxn5z5ekeqp2nzlfnumjk8l7uulen8rc0kaur9gscy9fq8 Amount: 9.000000000000 txch (9000 mojo), Confirmed in block: 1930591
```

Where clearly 9 TXCH does not equal 9000 mojos.

This PR fixes the display to use the word CAT instead of XCH/TXCH

```
Coin ID: 0xa7a087b1d3f49937dcada65f343a0797ebc75c39176c58629689e426355a1baa
	Address: txch1fckd972smrl3pxn5z5ekeqp2nzlfnumjk8l7uulen8rc0kaur9gscy9fq8 Amount: 9.000000000000 CAT (9000 mojo), Confirmed in block: 1930591
```

This is a fix for a new feature not yet released

### CODE REVIEW

**Review by wallentx — DISMISSED** [review]

**Review by altendky — CHANGES_REQUESTED** [review]
I thought we had code to output the proper CAT ticker/name in the CLI.  Also, a CAT amount shouldn't be shown with 12 decimal places, but instead three.  Feel free to handle that separately though.

File: chia/cmds/coin_funcs.py [hunk 1]
```diff
@@ -71,12 +78,18 @@ async def async_list(args: Dict[str, Any], wallet_client: WalletRpcClient, finge
...
         )
 
 
 def print_coins(
-    target_string: str, coins: List[Tuple[Coin, str]], mojo_per_unit: int, addr_prefix: str, paginate: bool
+    target_string: str,
+    coins: List[Tuple[Coin, str]],
+    mojo_per_unit: int,
+    addr_prefix: str,
+    coin_type: str,
```
Comment (altendky) [inline chia/cmds/coin_funcs.py:91]: This seems more like the ticker or name than the 'type'?

File: chia/cmds/coin_funcs.py [hunk 2]
```diff
@@ -44,6 +45,10 @@ async def async_list(args: Dict[str, Any], wallet_client: WalletRpcClient, finge
         excluded_amounts=final_excluded_amounts,
         excluded_coin_ids=excluded_coin_ids,
     )
+    coin_type: str = "XCH" if addr_prefix == "xch" else "TXCH"
+    if wallet_type == WalletType.CAT:
+        coin_type = "CAT"  # override XCH/TXCH prefix for CAT wallet
```
Comment (altendky) [inline chia/cmds/coin_funcs.py:50]: https://github.com/Chia-Network/chia-blockchain/blob/d13f8b837fd585c122b4c7366bf723aea1d4affd/chia/cmds/show_funcs.py#L131

I'd probably prefer an `if`/`elif`/`else` over picking the distinctly wrong thing first.

```suggestion
    if wallet_type == WalletType.XCH:
        coin_type: str = "XCH" if addr_prefix == "xch" else "TXCH"
    elif wallet_type == WalletType.CAT:
        coin_type = "CAT"  # override XCH/TXCH prefix for CAT wallet
    else:
        <something, i dunno what we want to do>
```

This is also relevant I think.

https://github.com/Chia-Network/chia-blockchain/blob/d13f8b837fd585c122b4c7366bf723aea1d4affd/chia/cmds/wallet_funcs.py#L87-L100

File: chia/cmds/coin_funcs.py [hunk 3]
```diff
@@ -26,7 +27,7 @@ async def async_list(args: Dict[str, Any], wallet_client: WalletRpcClient, finge
     if paginate is None:
         paginate = sys.stdout.isatty()
     try:
-        wallet_type = await get_wallet_type(wallet_id=wallet_id, wallet_client=wallet_client)
+        wallet_type: WalletType = await get_wallet_type(wallet_id=wallet_id, wallet_client=wallet_client)
```
Comment (altendky) [inline chia/cmds/coin_funcs.py:30]:
```suggestion
        wallet_type = await get_wallet_type(wallet_id=wallet_id, wallet_client=wallet_client)
```

File: chia/cmds/coin_funcs.py (hunk 2 above)
Comment (emlowe) [reply chia/cmds/coin_funcs.py:50]: yea, I worried about getting the entire CAT name to display here. since it could be quite long to display in each coin

Comment (emlowe) [reply chia/cmds/coin_funcs.py:50]: This is also why I went with the simpler form of naming it first - it should be XCH/TXCH in most cases, except only for CAT wallets so an alternative would be 
```
    if wallet_type == WalletType.CAT:
        coin_type = "CAT"  # override XCH/TXCH prefix for CAT wallet
    else 
        coin_type: str = "XCH" if addr_prefix == "xch" else "TXCH"
```

File: chia/cmds/coin_funcs.py (hunk 1 above)
Comment (emlowe) [reply chia/cmds/coin_funcs.py]: changed variable name

File: chia/cmds/coin_funcs.py (hunk 2 above)
Comment (emlowe) [reply chia/cmds/coin_funcs.py]: changed

File: chia/cmds/coin_funcs.py (hunk 3 above)
Comment (emlowe) [reply chia/cmds/coin_funcs.py]: changed

**Comment by emlowe** [comment]
Closing due to removable of feature

**Pull request closed without merging**